azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_provider.py

```diff
@@ -17,7 +17,7 @@ class MQTTProvider(object):
     to publish/subscribe messages.
     """
 
-    def __init__(self, client_id, hostname, username, password, ca_cert=None):
+    def __init__(self, client_id, hostname, username, password, x509_info = None, ca_cert=None):
```

Comment:
>x509_info = None [](start = 64, length = 16)

Should the x509_info parameter be on the connect function because that's where it's used?

For SAS renewal, I moved the password to connect (and reconnect) because that makes the upper layer more logical and saves us from caching the credentials down here.  

FWIW, I think the hostname and username should maybe also move to the connect method for the same reason, but I wanted to limit my delta.